apps/settings/js/panels/operator_settings/models/network_type_manager.js

```diff
@@ -0,0 +1,88 @@
+/* global getSupportedNetworkInfo */
+define(function(require) {
+  'use strict';
+
+  var Module = require('modules/base/module');
+  var Observable = require('modules/mvvm/observable');
+  var ObservableArray = require('modules/mvvm/observable_array');
+  var SettingsHelper = require('shared/settings_helper');
+
+  const SETTINGS_KEY = 'ril.radio.preferredNetworkType';
+
+  var NetworkTypeManager = Module.create(function NetworkTypeManager(conn) {
+    if (!conn.setPreferredNetworkType) {
+      this.throw(`mobile connection does not support settings preferred network
```

Comment:
And I don't understand why we have to add this check here ? Under what situation would make us unable to get this API ?